src/pt/sakuramangas/src/eu/kanade/tachiyomi/extension/pt/sakuramangas/SakuraMangas.kt

```diff
@@ -0,0 +1,331 @@
+        val mangaId = client.newCall(chapterListRequest(manga))
+            .execute()
+            .asJsoup()
+            .selectFirst("meta[manga-id]")!!
+            .attr("manga-id")
+
+        var page = 1
+        val chapters = mutableListOf<SChapter>()
+        do {
+            val response = client.newCall(chapterListApiRequest(mangaId, page++)).execute()
+            val chapterGroup = chapterListParse(response).also {
+                chapters += it
+            }
+        } while (chapterGroup.isNotEmpty())
+
+        return Observable.just(chapters)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+
+        return document.select(".capitulo-item").map { element ->
+            SChapter.create().apply {
+                name = buildString {
+                    element.selectFirst(".num-capitulo")
+                        ?.text()
+                        ?.let { append(it) }
+
+                    element.selectFirst(".cap-titulo")
+                        ?.text()
+                        ?.takeIf { it.isNotBlank() }
+                        ?.let { append(" - $it") }
+                }
+                scanlator = element.selectFirst(".scan-nome")?.text()
+                chapter_number =
+                    element
+                        .selectFirst(".num-capitulo")!!
+                        .attr("data-chapter")
+                        .toFloatOrNull() ?: 1F
+                date_upload = element.selectFirst(".cap-data")?.text()?.toDate() ?: 0L
+                setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+            }
+        }
+    }
+
+    // ================================ Pages =======================================
+
+    private fun pageListApiRequest(chapterId: String, token: String): Request {
+        val form = FormBody.Builder()
+            .add("chapter_id", chapterId)
+            .add("token", token)
+
+        return POST(
+            "$baseUrl/dist/sakura/models/capitulo/capitulos_read.php",
+            headers,
+            form.build(),
+        )
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+
+        val chapterId = document.selectFirst("meta[chapter-id]")!!.attr("chapter-id")
+        val token = document.selectFirst("meta[token]")!!.attr("token")
+
+        val response = client.newCall(pageListApiRequest(chapterId, token)).execute()
+            .parseAs<SakuraMangaChapterReadDto>()
+
+        val baseUrl = document.baseUri().trimEnd('/')
+
+        return response.imageUrls.mapIndexed { index, url ->
+            Page(
+                index,
+                imageUrl = "$baseUrl/$url".toHttpUrl().toString(),
+            )
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun getFilterList(): FilterList {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
```suggestion
        thread {
```

And you should check if it's already fetched.